### PR TITLE
Nexodus cable driver

### PR DESCRIPTION
**Pull request by sridhargaddam**
This WIP PR can be used to try out nexodus as one of the 
cable-drivers in Submariner.

To try this PR, one has to build the associated submariner-gateway
and submariner-routeagent images, push them to some private repo and 
use `image-override` options while joining the cluster. 
Also, the cable-driver should be explicitly specified as `nexodus`.

```
subctl join --cable-driver nexodus --image-override="submariner-gateway=quay.io/sridhargaddam/submariner-gateway:nexodus" --image-override=submariner-routeagent=quay.io/sridhargaddam/submariner-route-agent:nexodus ./broker-info.subm
```

### CODE REVIEW

File: pkg/routeagent_driver/handlers/ovn/host_networking.go [hunk 1]
```diff
@@ -146,3 +175,21 @@ func (ovn *Handler) getNextHopOnK8sMgmtIntf() (*net.IP, error) {
...
+
+func (ovn *Handler) getIPv4AddrOnK8sMgmtIntf() (*net.IP, error) {
+	link, err := netlink.LinkByName(OVNK8sMgmntIntfName)
+	if err != nil && !errors.Is(err, netlink.LinkNotFoundError{}) {
+		return nil, errors.Wrapf(err, "error retrieving link by name %q", OVNK8sMgmntIntfName)
+	}
+
+	addrs, err := netlink.AddrList(link, netlink.FAMILY_V4)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get the IP4 address list for interface %q", OVNK8sMgmntIntfName)
```
Comment (nyechiel) [inline pkg/routeagent_driver/handlers/ovn/host_networking.go]:
```suggestion
		return nil, errors.Wrapf(err, "failed to get the IPv4 address list for interface %q", OVNK8sMgmntIntfName)
```

**Comment by nyechiel** [comment]
bump

**Comment by tpantelis** [comment]
Is this still relevant?

**Comment by sridhargaddam** [comment]
This PR includes the high level changes to get a working POC with Nexodus cable-driver. However, both @mkolesnik and I feel that the proper way to support Nexodus would be to implement it as a cable-driver in Submariner. 

Lets close this PR. We can refer to this PR as a reference while adding proper support.

Related Shipyard PR: https://github.com/submariner-io/shipyard/pull/1273

**Pull request closed without merging**